lib/logstash/outputs/webhdfs.rb

```diff
@@ -0,0 +1,284 @@
+# Date: 2014-05-21
+#
+# Copyright 2014 dbap GmbH.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+require "logstash/namespace"
+require "logstash/outputs/base"
+require "stud/buffer"
+
+# Summary: Plugin to send logstash events to to files in HDFS via webhdfs 
+# restapi.  
+#
+# This plugin only has a mandatory dependency on the webhdfs gem from 
+# Tagamori Satoshi (@see: https://github.com/kzk/webhdfs).
+# Optional dependencies are zlib and snappy gem. 
+# No jars from hadoop are needed, thus reducing configuration and compatibility
+# problems.
+#
+# USAGE:
+# This is an example of logstash config:
+#
+# webhdfs {
+#   server => "127.0.0.1:50070"         (required)
+#   path => "/user/logstash/dt=%{+YYYY-MM-dd}/logstash-%{+HH}.log"  (required)
+#   user => "hue"                       (optional)
+#   message_format => "%{@source_host}" (optional)
+#   idle_flush_time => 10               (optional)
+#   flush_size => 50                    (optional)
+#   open_timeout => 15                  (optional)
+#   read_timeout => 15                  (optional)
+#   use_httpfs => true                  (optional)
+#   retry_interval => 1                 (optional)
+#   retry_times => 3                    (optional)
+#   compress => "snappy"                (optional)
+#   remove_at_timestamp => false        (optional)
+# }
+
+class LogStash::Outputs::WebHdfs < LogStash::Outputs::Base
+  include Stud::Buffer
+
+  config_name "webhdfs"
+  milestone 1
+
+  if RUBY_VERSION[0..2] == '1.8'
+    MAGIC = "\x82SNAPPY\x0"
+  else
+    MAGIC = "\x82SNAPPY\x0".force_encoding Encoding::ASCII_8BIT
+  end
+  DEFAULT_VERSION = 1
+  MINIMUM_COMPATIBLE_VERSION = 1
+
+  # The server name and port for webhdfs/httpfs connections.
+  config :server, :validate => :string, :required => true
+
+  # The Username for webhdfs.
+  config :user, :validate => :string, :required => false
+
+  # The path to the file to write to. Event fields can be used here,
+  # as well as date fields in the joda time format, e.g.:
+  # "/user/logstash/dt=%{+YYYY-MM-dd}/%{@source_host}-%{+HH}.log"
+  config :path, :validate => :string, :required => true
+
+  # The format to use when writing events to the file. This value
+  # supports any string and can include %{name} and other dynamic
+  # strings.
+  #
+  # If this setting is omitted, the full json representation of the
+  # event will be written as a single line.
+  config :message_format, :validate => :string
```

Comment:
Done in https://github.com/dstore-dbap/logstash-output-webhdfs-1/commit/8f1cf04a26eeadf9c448ae7da90af5af8ffee220